File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/ApiKeyApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class ApikeyApi {
    private final HttpClient client;
    
    public ApikeyApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing API key */
    public PlusApiResultPlusApiKeyVO update(PlusApiKeyForm body) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.put(ApiPaths.backendPath("/apikey"), body);
    }

    /** Create a new API key */
    public PlusApiResultPlusApiKeyVO create(PlusApiKeyForm body) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey"), body);
    }

    /** Get API key detail for current user */
    public PlusApiResultPlusApiKeyVO getMyApiKey(String id) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.get(ApiPaths.backendPath("/apikey/my/" + id + ""));
    }

    /** Update API key for current user */
    public PlusApiResultPlusApiKeyVO updateMyApiKey(String id, PlusApiKeySelfUpdateForm body) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.put(ApiPaths.backendPath("/apikey/my/" + id + ""), body);
    }

    /** Delete API key for current user */
    public PlusApiResultBoolean deleteMyApiKey(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/apikey/my/" + id + ""));
    }

    /** Create API key for current user */
    public PlusApiResultPlusApiKeyVO createMyApiKey(PlusApiKeySelfCreateForm body) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/my"), body);
    }

    /** Rotate API key for current user */
    public PlusApiResultPlusApiKeyVO rotateMyApiKey(String id) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/my/" + id + "/rotate"), null);
    }

    /** Enable API key for current user */
    public PlusApiResultPlusApiKeyVO enableMyApiKey(String id) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/my/" + id + "/enable"), null);
    }

    /** Disable API key for current user */
    public PlusApiResultPlusApiKeyVO disableMyApiKey(String id) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/my/" + id + "/disable"), null);
    }

    /** Get API keys by page */
    public PlusApiResultPagePlusApiKeyVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/list"), body, params);
    }

    /** Get all API keys */
    public PlusApiResultListPlusApiKeyVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusApiKeyVO) client.post(ApiPaths.backendPath("/apikey/list/all"), body);
    }

    /** Get an API key by ID */
    public PlusApiResultPlusApiKeyVO getById(String id) throws Exception {
        return (PlusApiResultPlusApiKeyVO) client.get(ApiPaths.backendPath("/apikey/" + id + ""));
    }

    /** Delete an API key */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/apikey/" + id + ""));
    }

    /** List API keys for current user */
    public PlusApiResultListPlusApiKeyVO listMyApiKeys() throws Exception {
        return (PlusApiResultListPlusApiKeyVO) client.get(ApiPaths.backendPath("/apikey/my/list"));
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/ApiKeyApi.swift
import Foundation

public class ApikeyApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing API key
    public func update(body: PlusApiKeyForm) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.put(ApiPaths.backendPath("/apikey"), body: body)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Create a new API key
    public func create(body: PlusApiKeyForm) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey"), body: body)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Get API key detail for current user
    public func getMyApiKey(id: String) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.get(ApiPaths.backendPath("/apikey/my/\(id)"))
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Update API key for current user
    public func updateMyApiKey(id: String, body: PlusApiKeySelfUpdateForm) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.put(ApiPaths.backendPath("/apikey/my/\(id)"), body: body)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Delete API key for current user
    public func deleteMyApiKey(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/apikey/my/\(id)"))
        return response as? PlusApiResultBoolean
    }

    /// Create API key for current user
    public func createMyApiKey(body: PlusApiKeySelfCreateForm) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/my"), body: body)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Rotate API key for current user
    public func rotateMyApiKey(id: String) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/my/\(id)/rotate"), body: nil)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Enable API key for current user
    public func enableMyApiKey(id: String) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/my/\(id)/enable"), body: nil)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Disable API key for current user
    public func disableMyApiKey(id: String) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/my/\(id)/disable"), body: nil)
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Get API keys by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusApiKeyVO
    }

    /// Get all API keys
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusApiKeyVO? {
        let response = try await client.post(ApiPaths.backendPath("/apikey/list/all"), body: body)
        return response as? PlusApiResultListPlusApiKeyVO
    }

    /// Get an API key by ID
    public func getById(id: String) async throws -> PlusApiResultPlusApiKeyVO? {
        let response = try await client.get(ApiPaths.backendPath("/apikey/\(id)"))
        return response as? PlusApiResultPlusApiKeyVO
    }

    /// Delete an API key
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/apikey/\(id)"))
        return response as? PlusApiResultBoolean
    }

    /// List API keys for current user
    public func listMyApiKeys() async throws -> PlusApiResultListPlusApiKeyVO? {
        let response = try await client.get(ApiPaths.backendPath("/apikey/my/list"))
        return response as? PlusApiResultListPlusApiKeyVO
    }
}
